keps/sig-node/277-ephemeral-containers/README.md

```diff
@@ -221,28 +220,33 @@ There are no limits on the number of Ephemeral Containers that can be created in
 a pod, but exceeding a pod's resource allocation may cause the pod to be
 evicted.
 
-### Identifying Pods with Ephemeral Containers
+### Reattaching Ephemeral Containers
 
-The kubelet will set a `PodCondition` when it starts an Ephemeral Container.
-This condition may not be cleared: it will exist for the lifetime of the Pod
-and continues to exist even if all Ephemeral Containers are removed.
+One may reattach to a Ephemeral Container using `kubectl attach`. When supported
+by a runtime, multiple clients can attach to a single debug container and share
+the terminal. This is supported by the Docker runtime.
 
-The intended use of this `PodCondition` is to enable administrators to enforce
-custom policies for pods that have had Ephemeral Containers. For example,
-cluster administrators may want to automatically apply a label or delete the pod
-after a configurable time. This may be accomplished by a controller watching
-for this `PodCondition`, though the implementation of such a controller is out
-of scope for this proposal.
+### Ephemeral Container Lifecycle
 
-### Reattaching and Restarting Ephemeral Containers
+Ephemeral containers will stop when their command exits, such as exiting a
+shell, and they will not be restarted.  Unlike `kubectl exec`, processes in
+Ephemeral Containers will not receive an EOF if their connections are
+interrupted, so shells won't automatically exit on disconnect.
 
-One can reattach to a Ephemeral Container using `kubectl attach`. When supported
-by a runtime, multiple clients can attach to a single debug container and share
-the terminal. This is supported by Docker.
+There is no API support for killing or restarting an ephemeral container.
+The only way to exit the container is to send it an OS signal.
 
-Ephemeral Containers will not be restarted automatically, and there is no
-method in the API to restart an Ephemeral Container.  Creators of Ephemeral
-Containers are expected to choose a new, unused name.
+### Removing Ephemeral Containers
+
+Ephemeral containers may not be removed from a Pod once added, so the
+`pod.spec.ephemeralContainers` field will contain a complete history
+and may be used to identify whether a pod has ever contained an ephemeral
+container.
+
+We've received feedback during the alpha period from users that would like
+to be able to remove ephemeral containers (see [#84764](https://issues.k8s.io/84764)).
+This is out of scope for the initial graduation of ephemeral containers,
+but it may be explored in a future KEP.
```

Comment:
Can we clarify what will be considered in the future? The goals of https://issues.k8s.io/84764 seem reasonable to address in the future (release resources, stop a running ephemeral container, etc), but if the API graduates to beta with:

> the `pod.spec.ephemeralContainers` field will contain a complete history and may be used to identify whether a pod has ever contained an ephemeral container

clients that use the API that way should be able to rely on that in the future

Reply:
According to https://github.com/kubernetes/kubernetes/pull/103354#issuecomment-897451068,  
> Should we be more prescriptive in the behavior we allow for ephemeral containers? Is there a place in the security policy controller for "allow removal of ephemeral containers" flag?

In my first view, ephemeral container is similar to `exec` and `complete history for ephemeral containers` is something like `command history in container`. 

Several ways 
1. feature gate like `AllowEphemeralContainersRemoving` 
2. A PodSecurity policy to prevent removing ephemeral containers(default allow). Or vice versa
3. adding a `state/status` to EphemeralContainer like `state=removed` (kubelet will remove the exited container if the state is removed. Skip it when running new ephemeral container)